test/es-module/test-esm-loader.mjs

```diff
@@ -43,6 +43,14 @@ await assert.rejects(
   { code: 'ERR_INVALID_RETURN_PROPERTY_VALUE' },
 );
 
+await assert.rejects(
+  import('esmHook/commonJsNullSource.mjs'),
+  {
+    code: 'ERR_INVALID_RETURN_PROPERTY_VALUE',
+    message: /for the "source" from the "load" function but got null/,
```

Comment:
It's generally less brittle to match key words within a human-friendly message so the test is sufficiently tolerant of non-substantive changes/word-smithing. Here I would have 3 matches looking for `source`, `load` and `null`: those are the critical pieces of info the user will need to identify the problem. They cannot change without materially changing the behaviour, so that's all we care about.

Case in point: the error message _should_ say `"load" hook` (instead of `"load" function`); that's not strictly in-scope for this PR, so you could toss that back. Then when I subsequently correct function → hook, this test as written would erroneously break (creating make-work), but if it were matching as I described above, the test would correctly continue to pass.

So instead of `assert.rejects`, I would catch the error so you have repeat access available to its message for those 3 `assert.match`s.